pkg/entrypoint/startup/startup.go

```diff
@@ -40,6 +42,14 @@ func runStartup(ctx context.Context, log *logrus.Entry) error {
 		return err
 	}
 
+	ns, err := startup.GetNameserversFromDNSConfig("/host")
+	if err != nil {
+		return err
+	}
+	if !reflect.DeepEqual(ns, cs.Properties.NetworkProfile.InUseNameservers) {
+		log.Warnf("overriding InUseNameservers with values from host resolver difference %s", cmp.Diff(ns, cs.Properties.NetworkProfile.InUseNameservers))
```

Comment:
this to me is a bit troubling what to do if the vnet and resolver come up with different values..

Reply:
yes; I think you're doing the right thing and I think it's the best that can be done.